cmake/Gems.cmake

```diff
@@ -8,6 +8,8 @@
 
 # This file contains utility wrappers for dealing with the Gems system.
 
+set(GEM_VARIANT_DEFAULT_HeadlessServers Servers)
+
```

Comment:
I recommend variable this logic into the `launcher_generator.cmake` and storing it into a global property.

The Gems.cmake file shouldn't have any knowledge about variant types and it is outside the responsiblity of this file to have that knowledge.

That should be configured in the CMake scripts that setup the Server, Clients, Unified launchers